docs/design/2025-03-17-sys-session-manage-enhancement.md

```diff
@@ -0,0 +1,173 @@
+    se, err := getFromPool()
+    if err != nil {
+        return nil, err
+    }
+    // register the internal session to the map
+    infosync.StoreInternalSession(se.GetSessionContext())
+    return se, nil
+}
+
+func (p *Pool) put(s *Session) {
+    // rollback the session first to reset the internal state.
+    s.Rollback()
+    // unregister the internal session from the map
+    infosync.DeleteInternalSession(s.GetSessionContext())
+    // put the session back to the pool
+    p.putToPool(s)
+}
+```
+
+The above code has been simplified for demonstration purposes. The session pool stores the internal session in a map when `Pool.get` is called and removes it from the map in `Pool.put`. Additionally, before being returned to the pool, the session is automatically rolled back to ensure its internal state is reset.
+
+With the newly implemented session pool that manages the session lifecycle, developers no longer need to worry about the session's state or lifecycle. 
+
+### Session
+
+The type `Session` is not a "real" one but a wrapper of the internal session, see:
+
+```go
+// Session is public to callers.
+type Session struct {
+    internal *session
+}
+```
+
+It provides a set of methods that proxy access to the internal session. When the caller invokes `WithSession`, a new `Session` instance is created and passed to the function. Once the function completes, the `Session` instance is destroyed, and only the internal session is returned to the pool.
+
+We have several reasons for proposing the above design:
+
+- The wrapped methods allow for additional checks to prevent misuse of the internal session. For example, if a caller attempts to invoke methods after the session has been returned to the pool, we can detect this and return an error.
+- The `Session` type can be extended with new methods, providing more convenient functionality for internal use.
+- The original session contains numerous methods, many of which may not be necessary for the caller. By filtering out unnecessary methods, we can expose only the required ones, simplifying the interface and improving usability.
+
+The internal session is not exposed to the caller and can only be accessed by the `Session` or `Pool` types. It has several fields, see:
+
+```go
+// session is private and can only be accessed by the `Pool` and `Session`.
+type session struct {
+    mu sync.Mutex
+    owner any
+    sctx sessionctx.Context
+    ...
+}
+```
+
+The `mu` protects the internal session from concurrent access, and the field `sctx` is the real session context that holds the session's state. 
+
+Let's provide further clarification on the `owner` field. The "owner" refers to the entity that holds the internal session, and only the owner is permitted to use it. When an internal session is in the pool, the pool itself is considered the owner. When a caller retrieves the session from the pool, the wrapped `Session` becomes the new owner, which is then passed to the caller.
```

Comment:
So the type of `owner` is any because it may be the pool or `Session`? 

Will it be `nil` `owner`? If not, I think we can change the type to `*Session` and let nilness representing it's owned by pool